spiceflow/src/listen-for-node.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  type Server,
+  type IncomingMessage,
+  type ServerResponse,
+  createServer,
+} from 'node:http'
+import { AddressInfo } from 'node:net'
+import { type Spiceflow, SpiceflowRequest } from './spiceflow.ts'
+import { superjsonSerialize } from './serialize.ts'
+
+export async function listenForNode(
+  app: Spiceflow<any, any, any, any, any, any>,
+  port: number,
+  hostname: string = '0.0.0.0',
+): Promise<Server<typeof IncomingMessage, typeof ServerResponse>> {
+  const server = createServer((req, res) => {
+    return handleNodeRequest(app, req, res)
+  })
+
+  await new Promise((resolve) => {
+    server.listen(port, hostname, () => {
+      // We could print from what we take as arguments of `serve`, but by reading
+      // the `server` object, we can ensure that they are properly set.
+      const address = server.address() as AddressInfo
+      console.log(`Listening on http://${address.address}:${address.port}`)
```

Comment:
@remorses Changed from `localhost` to `${address.address}`, which I believe is more accurate. The consequence is that it most often shows "Listening on http://0.0.0.0:3000" instead of "Listening on http://localhost:3000". If `localhost` is deemed more user-friendly, we can implement a special case: when `address.address` is `0.0.0.0`, show `localhost` instead.

Let me know what you think.